### ISSUE TITLE
Off by one error in Sources list.

### ISSUE THREAD

**Issue by CaptainApathy** [hhannine/superpaper#158]
Discrepancy between selected image in Browser, preview, and actual image applied to desktop.

Add 3 or more images to the source list in advanced span.
Select an image and hit apply.
Image in the preview is the one next in the list, actual image applied is the previous in the list.
Behaviour is inconsistent and can change how it is wrong depending on number of images in list.

**Expected behavior**
Selected image, previewed image, and applied image should all be the same image.

**Screenshots**
Screenshot was taken immediately after hitting apply. Second image selected, third image in preview, actual wallpaper on desktop is the first.
![Screenie](https://imgur.com/a/JZejN4t)

**Desktop (please complete the following information):**
 - OS: Currently Windows 11, also did it on ubuntu 22.04 with kde
 - Superpaper package & version: Windows installer 2.2.1

**Additional context**

Logging / debugging contents:
```
Dont know where to find logs, but can upload them if needed.
```

**Comment by hhannine** [comment]
Yes this is a known issue. It's related to how the randomized list is stored, can't quite remember anymore. I did try to workaround to solve this but couldn't figure it out then.

I'd like to solve it but I'm quite busy, help would be appreciated.
